classes/collector/base.php

```diff
@@ -53,6 +53,9 @@ public function record_metrics(array $metrics, \progress_bar $progress = null) {
                 $count++;
             }
         }
+        if ($count > 0) {
+            set_config($metric->name . '_last_value', $metrics[$count - 1]->value, 'tool_cloudmetrics');
```

Comment:
If more than one collector is on then this will run multiple times.

Conversely if no collectors are on then it won't run either, feels like this should be moved up a layer.

Reply:
Moved setting the config to the task